### ISSUE TITLE
dask chunking tutorial outline

### ISSUE THREAD

**Issue by dcherian** [xarray-contrib/xarray-tutorial#157]
from the pangeo working meeting discussion with @mgrover1 @jmunroe @norlandrhagen 

Here's an outline for an intermediate tutorial talking about dask chunking specifically for Xarray users 

-----
### Motivation: why care about chunk size?
- demonstrate relation between chunk size and computation time / number of tasks with a simple example?
  - maybe even memory usage
- https://tutorial.dask.org/02_array.html#Choosing-good-chunk-sizes
- https://docs.dask.org/en/stable/array-chunks.html

### Keeping track
- monitoring chunk sizes and num tasks throughout the pipeline using the repr
  - use some images

### Why is it important to choose appropriate chunks early in the pipeline?
- while output blocks may be small, intermediate blocks need not be.
- So keep monitoring chunksizes (and tasks) throughout the pipeline.
- Demonstrate that rechunking is not cheap in most cases

### Specify chunks when reading data
1. Avoid `chunks="auto"`.
   - https://docs.dask.org/en/stable/array-chunks.html#automatic-chunking
2. Specifying `chunks` during data read
   - `open_dataset`
   - `open_mfdataset`
3. Analysis vs storage chunks:
   - Dask chunks should be a multiple of chunks on disk
   - talk about aligning chunks with files stored on disk
   - David Hoese [example](https://github.com/pydata/xarray/discussions/6945#discussioncomment-3468286)

**Comment by scottyhq** [comment]
Agreed this would be great to document thoroughly. See also this relevant issue + discussion in rioxarray https://github.com/corteva/rioxarray/issues/253

**Comment by djhoese** [comment]
> demonstrate relation between chunk size and computation time / number of tasks with a simple example?
>   * maybe even memory usage

This would be huge! This comes up often in Satpy where users want to process satellite images on their local machine but they only have 8GB or 16GB of memory. If someone can make a good diagram showing chunks being processed by a worker thread/process and how changing the size of all chunks or number of workers contributes to the overall memory usage that would be such a help when explaining this to users.

**Comment by dcherian** [comment]
forgot to cc @rybchuk!
